### PR TITLE
[14.0] [MIG] stock_picking_batch_extended: Migrate to version 14.0

### PR DESCRIPTION
**Pull request by sonhd91**
* `stock_move.py` reassigned picking has `state=confirm` to `state=assigned` due `[product_id.type != "product"](https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1168)` which made test fail
* can't cancel multi batch becase [self.ensure_one()](https://github.com/odoo/odoo/blob/14.0/addons/stock_picking_batch/models/stock_picking_batch.py#L173) so remove test case
* review `test_package`: `stock_move.py` reassigned picking to `state=assigned` due `[product_id.type != "product"]` [link](https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1168), and we can't update `location_id` due to move_lines_id has `product_id.type='product'` [here](https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move_line.py#L497)

### CODE REVIEW

File: stock_picking_batch_extended/data/batch_picking_actions_server.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2018 Tecnativa - Carlos Dauden
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo noupdate="0">
```
Comment (leemannd) [inline stock_picking_batch_extended/data/batch_picking_actions_server.xml]: AFAIR `noupdate=0` is not usefull

**Review by leemannd — APPROVED** [review]

**Review by ajaniszewska-dev — APPROVED** [review]

**Review by ivantodorovich — APPROVED** [review]

File: stock_picking_batch_extended/data/batch_picking_actions_server.xml (hunk 1 above)
Comment (sonhd91) [reply stock_picking_batch_extended/data/batch_picking_actions_server.xml]: Thanks for review

**Comment by santostelmo** [comment]
@sonhd91 we have the following test falling:

```
ERROR: TestBatchPicking.test_partial_done
Traceback (most recent call last):
  File "/home/travis/build/OCA/stock-logistics-workflow/setup/stock_picking_batch_extended/odoo/addons/stock_picking_batch_extended/tests/test_batch.py", line 386, in test_partial_done
    backorder_confirmation_context = inmediate_transfer_action.get("context")
AttributeError: 'bool' object has no attribute 'get'
```

File: stock_picking_batch_extended/tests/test_batch.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+            .save()
+            .process()
+        )
+        backorder_confirmation_context = inmediate_transfer_action.get("context")
+        # Create backorder? action
+        Form(
+            self.env["stock.backorder.confirmation"].with_context(
+                backorder_confirmation_context
+            )
+        ).save().process()
```
Comment (santostelmo) [inline stock_picking_batch_extended/tests/test_batch.py]: @sonhd91 to adapt/fix the unit test falling I suggest:
```suggestion
        self.picking2.move_line_ids[0].qty_done = 1
        self.picking2.move_line_ids[1].qty_done = 1
        context = self.batch.action_transfer().get("context")
        # Create backorder? action
        Form(
            self.env["stock.backorder.confirmation"].with_context(
                context
            )
        ).save().process()
```

Comment (sonhd91) [reply stock_picking_batch_extended/tests/test_batch.py]: Thanks for review, I just update the branch

**Review by santostelmo — APPROVED** [review]

**Pull request merged**